### PR TITLE
Node-RED state machine

### PR DESCRIPTION
**Pull request by Scarjit**
Closes https://github.com/united-manufacturing-hub/MgmtIssues/issues/1613
Closes https://github.com/united-manufacturing-hub/MgmtIssues/issues/1614

### CODE REVIEW

File: umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+Before producing anything our system first needs to know what we are producing. This is done by creating a product-type. In this example we will create a product-type called "otto-poster" (Prints of our beloved mascot Otto).
+Therefore, we create an inject node and a mqtt node (to publish the product-type to the UNS).
+Since we are dealing with a printer, we use a low cycle time of 10 milliseconds.
+
+1. Drag an inject node from the palette to the flow.
+2. Double-click the inject node and set the payload to:
+```json
+{
+  "externalProductID": "otto-poster",
+  "cycleTime": 10
```
Comment (JeremyTheocharis) [inline umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md:30]: time is here in seconds :)

File: umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+```
+6. Drag an mqtt out node from the palette to the flow.
+7. Configure the mqtt node to use `united-manufacturing-hub-mqtt` as the Server, you can leave the topic empty as we have already set it in the inject node.
+8. Connect the inject node to the function node and the function node to the mqtt node.
+9. Deploy the flow.
+
+
+### Step 5: Creating the state machine
+
+This step is quite complex. We will create a state machine to simulate the printing process.
```
Comment (JeremyTheocharis) [inline umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md:116]: This looks more complicated than it is. How about that we start with two inject nodes "start machine" / "stop machine" just to show how states work. Then maybe one additional one for "produce product" (and note that one could now theoretically derive the machine state from whenever a product was produced).

"Normally you get the machine state directly from the machine. In the first step, we will simply inject a "machine running", and "machine stopped" with inject node. As a more advanced step, we will build a simulator in Node-RED.

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]
Let Sebastian & Mateusz try this out and see if they understands it

**Comment by JeremyTheocharis** [comment]
An in general, this looks quite good :)

**Review by DaruZero — CHANGES_REQUESTED** [review]
Styling:
- Avoid using `Step 1` in headings. Just the header name is fine
- Try to keep each line between 80-100 character, for easier reviewing
- See the [`codenew`](https://umh.docs.umh.app/docs/development/contribute/documentation/style/hugo-shortcodes/#code-example) shortcode for the nodered flow

**Comment by Ayato77** [comment]
-  Maybe, renaming function nodes makes clear which nodes are should be connected. For example, in step 54, one should connect delay node and function node created in step 17, but identifying the function node might be difficult. (Step 5: Creating the state machine)

- A picture(s) might help user to check the result.
- Maybe, mention that one should choose JSON as the input type of inject's payload.

**Comment by DaruZero** [comment]
Put the code snippets for the `codenew` shortcode in the `content/en/examples` folder. You can read how it works in [our documentation](https://umh.docs.umh.app/docs/development/contribute/documentation/style/hugo-shortcodes/#code-example)

File: umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md [hunk 3]
```diff
@@ -0,0 +1,385 @@
+---
+title: 6. (optional) Calculating business KPIs such as OEE with _analytics
+menuTitle: 6. (optional) Calculating business KPIs such as OEE with _analytics
+description: Generate business KPIs such as OEE with _analytics
+weight: 6000
+draft: true
```
Comment (JeremyTheocharis) [inline umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md:6]: draft = false :)

File: umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md [hunk 4]
```diff
@@ -0,0 +1,385 @@
...
+we will simply inject a "machine running", and "machine stopped" with inject node. 
+As a more advanced step, we will build a simulator in Node-RED.
+
+#### Injecting running and stopped states
+
+In this step, we create a simple flow injecting running and stopped state of machine.
+
+1. Drag an inject node from the palette to the flow. This node injects the start status.
+2. Double-click the inject node and set the payload to:
+```json
```
Comment (JeremyTheocharis) [inline umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md:138]: When working with states we should use the states message, not the workorder message

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]

File: umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md (hunk 4 above)
Comment (Ayato77) [reply umh.docs.umh.app/content/en/docs/getstarted/calculatekpiwithanalytics.md:138]: Revised and simplified the flow

**Review by JeremyTheocharis — APPROVED** [review]

**Pull request merged**